src/clj/rems/db/applications.clj

```diff
@@ -954,16 +955,25 @@
   (update (cheshire/parse-string wf keyword)
           :type keyword))
 
-(defn- string->datetime [s]
+(defn- datestring->datetime [s]
+  (if (string? s)
+    (time-format/parse s)
+    s))
+
+(defn- longstring->datetime [s]
   (if (string? s)
     (time-coerce/from-long (Long/parseLong s))
     s))
 
 (def ^:private datetime-coercion-matcher
-  {DateTime string->datetime})
+  {DateTime longstring->datetime})
+
+(def ^:private datestring-coercion-matcher
+  {DateTime datestring->datetime})
 
 (defn- coercion-matcher [schema]
-  (or (datetime-coercion-matcher schema)
+  (or (datestring-coercion-matcher schema)
+      (datetime-coercion-matcher schema)
```

Comment:
Maybe better names for the prefixes would be  "iso-datetime" and "unix-datetime"